### ISSUE TITLE
Help | About | Translated By small bugs

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#583]
Original [issue 583](https://code.google.com/p/pychess/issues/detail?id=583) reported by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-09-02

**What steps will reproduce the problem?**

Help | About | Translated By
1. First line states that PyChess is translated into 33 languages, but I see there are only 29 directories in lang/ directory, so most probably this "33" number is incorrect.

My suggestion is to remove this first line, because this tab is not about statistics, but translators name. So I think it should not contain number of languages that PyChess is translated into.
1. I suggest to remove "translators" word and just use name of language instead or even to make it more clear that first line of language is actually language name and not translators name like:

=== Azerbaijani ===
Elxan İsmayılov

=== Arabic ===
Agari
Ahmad Farghal
1. There is no info about Slovenian translator (you now like me).  :)
2. English (United Kingdom) translators has "Me" - is this really the name of translator or is this just "me" as some name missing here.

**Did you use an installed version of PyChess or did you run in from a**
**tarball/svn?**
Latest hg.

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=583#c1) posted by [gbtami](https://code.google.com/u/gbtami/) 2010-09-02
- added label: Defect

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=583#c2) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-09-04

I did the following:
1. removed first line of "PyChess is translated into 33 languages", because this is about translators and not statistics and also every time new language is added this number should manually change in file - annoying. So just removed it.
2. I removed "translators"" word like "Arabic translators", so now it is only "Arabic"
3. Put character "_" before translation language, so Arabic now becomes "_ Arabic" - this separates languages from translators.
4. I added myself to "Slovenian" translators. :)

Please see attached file. If you like the change then please commit this new file.
- attached [TRANSLATORS](https://code.google.com/p/pychess/issues/detail?id=583#c2)

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=583#c3) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-09-04

gbtami, I see you have made some changes to translators.py file in revision d6394f1d6e. Now I understand TRANSLATORS file is automatically generated from Launchad.

But are you sure all of translators will be listed in TRANSLATORS file if only import from Launchpad is made? What about upstream translators like me?

Also importing translators from Launchpad will import translators from older version of PyChess. Not all of that translations are available in PyChess anymore - if you look into lang directory you will see there are some languages missing from TRANSLATORS file.

**Comment by pychess** [comment]
Original [comment 4.](https://code.google.com/p/pychess/issues/detail?id=583#c4) posted by [gbtami](https://code.google.com/u/gbtami/) 2010-09-04

Heh, that commit was made accidentally :) I'v just started to play with it.
Your points are correct, we have to rework this script.

**Comment by pychess** [comment]
Original [comment 5.](https://code.google.com/p/pychess/issues/detail?id=583#c5) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-09-11

If I understand correctly Thomas has made an update of script and now TRANSLATORS file looks updated with Launchpad now showing 50 languages. I also see that list is a merge of Launchpad and upstream PyChess translations shipped directly into Google code (like my translations). So this looks fixed.

Three more issues on this topic:
1. It looks like TRANSLATION file is not alphabetically ordered by translation languages any more, you see e.g. for Spanish:
English (United Kingdom)
Spanish
Estonian
but also whole ordering of languages is not alphabetically.
1. I also think that translator names inside individual language should be ordered alphabetically.
2. This TRANSLATOR file is not reflected in actual PyChess code in Help | About | Credits | Tranlated by tab. PyChess still has an old list. If I understand correctly PyChess is using (probably some manual import) from TRANSLATION file.

**Comment by pychess** [comment]
Original [comment 6.](https://code.google.com/p/pychess/issues/detail?id=583#c6) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-10-25

I have updated TRANSLATORS file with ./translators.py and commited changes: Rev: 1d4d00955c

It looks like PyChess is not reading TRANSLATORS file when PyChess is running. It looks there is some kind of import done into python code. Help | About still shows 33 languages.

**Comment by pychess** [comment]
Original [comment 7.](https://code.google.com/p/pychess/issues/detail?id=583#c7) posted by [gbtami](https://code.google.com/u/gbtami/) 2010-10-25

With revision 3b7e3c9146 the About dialog is reading data from AUTHORS, ARTISTS, DOCUMENTERS and TRANSLATORS files.

**Comment by pychess** [comment]
Original [comment 8.](https://code.google.com/p/pychess/issues/detail?id=583#c8) posted by [gbtami](https://code.google.com/u/gbtami/) 2010-10-26

The order issue is fixed in revision 4b6e4ca129.
- changed status to: Fixed

**Comment by pychess** [comment]
Original [comment 9.](https://code.google.com/p/pychess/issues/detail?id=583#c9) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-10-26

gbtami, it looks ok for 99,999% of users/languages, but for Slovenian is not correct sorting. I think the reason is because all of the users are acquired from launchpad except for me (Igor). In translators.py file there is extra string "extraTranslators" that is not sorted correctly.

**Comment by pychess** [comment]
Original [comment 10.](https://code.google.com/p/pychess/issues/detail?id=583#c10) posted by [gbtami](https://code.google.com/u/gbtami/) 2010-10-26

You are right :) To solve this in the shortest way I ask you to register to lunchpad, and touch the sl translation at least once. Then we can forget the "extraTranslators" workaround.

**Comment by pychess** [comment]
Original [comment 11.](https://code.google.com/p/pychess/issues/detail?id=583#c11) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-10-26

gbtami, excellent idea, why having a hack for one single exception. I have created Launchpad account, made some "touch" to the sl translation.

Then I have deleted TRANSLATORS file. Deleted extraTranslators strings from translators.py file and committed the changes to revision 503b197ed9.
